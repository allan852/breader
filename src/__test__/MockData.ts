import { IntlProvider } from 'react-intl'
import { messages } from '../locales'
import { IArticle, IFeed } from '../schemas'

export const intlProviderProps: IntlProvider.Props = {
  locale: 'en-US',
  messages: messages['en-US'],
}

const intlProvider = new IntlProvider(intlProviderProps, {})
export const intl = intlProvider.getChildContext().intl

export const feed: IFeed = {
  _id: 'http://www.ruanyifeng.com/blog/atom.xml',
  _rev: '5-55772bf2b4d4b74608a60ac3480334e4',
  author: 'ruanyifeng',
  categories: [],
  createTime: 1563686468437,
  deleteTime: 0,
  description: "Ruan YiFeng's Blog",
  etag: '"11d86-58e2cc1eab031"',
  favicon: 'http://www.ruanyifeng.com/favicon.ico',
  generator: 'Movable Type 5.2.2 (http://www.sixapart.com/movabletype/)',
  language: '',
  link: 'http://www.ruanyifeng.com/blog/',
  publishTime: 1563697765000,
  time: 1563697765000,
  title: '阮一峰的网络日志',
  url: 'http://www.ruanyifeng.com/blog/atom.xml',
}

export const article: IArticle = {
  _id: 'tag:www.ruanyifeng.com,2019:/blog//1.2100',
  _rev: '5-ff6fc450616071a8f7d022b7532e85aa',
  author: 'ruanyifeng',
  categories: [],
  comments: 'article comments',
  createTime: 1563697765000,
  deleteTime: 0,
  description: 'article description',
  enclosures: [],
  feedId: 'http://www.ruanyifeng.com/blog/atom.xml',
  image: '',
  isDayFirst: true,
  isStarred: false,
  isUnread: false,
  link: 'http://www.ruanyifeng.com/blog/2019/07/weekly-issue-65.html',
  originLink: '',
  publishTime: 1563697765000,
  summary: 'article summary',
  time: 1563697765000,
  title: 'article title',
}
