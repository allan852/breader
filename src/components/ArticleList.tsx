import { Icon, Modal, Radio } from 'antd'
import { RadioChangeEvent } from 'antd/lib/radio'
import Immutable from 'immutable'
import React, { Component } from 'react'
import { injectIntl, WrappedComponentProps } from 'react-intl'
import ArticleVirtualList from '../containers/ArticleVirtualList'
import SearchArticleModal from '../containers/SearchArticleModal'
import { EArticleFilter, EMenuKey, IArticle, IFeed } from '../schemas'
import ArticleListSkeleton from '../skeletons/ArticleListSkeleton'

import '../styles/ArticleList.less'

const RadioButton = Radio.Button
const RadioGroup = Radio.Group
const Confirm = Modal.confirm

export interface IArticleListOwnProps {}

export interface IArticleListStateProps {
  articles: Immutable.List<IArticle>
  articlesFilter: string
  feeds: Immutable.List<IFeed>
  isFetchingArticles: boolean
  selectedMenuKey: string
}

export interface IArticleListDispatchProps {
  asyncFilterArticles: (filter: string) => Promise<void>
  asyncSetAllArticlesRead: (ids: string[]) => Promise<number>
}

export interface IArticleListProps
  extends IArticleListOwnProps,
    IArticleListStateProps,
    IArticleListDispatchProps {}

export interface IArticleListState {
  isSearchArticleModalVisible: boolean
  chooseItemIndex: number
}

export class ArticleListComponent extends Component<
  IArticleListProps & WrappedComponentProps,
  IArticleListState
> {
  public constructor(props: IArticleListProps & WrappedComponentProps) {
    super(props)
    this.state = {
      chooseItemIndex: -1,
      isSearchArticleModalVisible: false,
    }
  }
  public handleRadioChange = (e: RadioChangeEvent) => {
    const target = e.target
    this.props.asyncFilterArticles(target.value as string)
  }
  public handleSearchClick = () => {
    this.setState({
      isSearchArticleModalVisible: true,
    })
  }
  public handleSearchCancel = () => {
    this.setState({
      isSearchArticleModalVisible: false,
    })
  }
  public handleCheckClick = (e: React.MouseEvent<HTMLElement, MouseEvent>) => {
    Confirm({
      onOk: () => {
        const ids = this.props.articles
          .filter(article => article.isUnread)
          .map(article => article._id)
          .toArray()
        this.props.asyncSetAllArticlesRead(ids)
      },
      title: this.props.intl.formatMessage({
        id: 'doYouWantSetAllArticlesBeRead',
      }),
    })
  }
  public handleSearchItemChoose = (index: number) => {
    this.setState({
      isSearchArticleModalVisible: false,
    })
    if (index > -1 && index !== this.state.chooseItemIndex) {
      this.setState({
        chooseItemIndex: index,
      })
    }
  }

  public render() {
    const {
      articlesFilter,
      feeds,
      isFetchingArticles,
      selectedMenuKey,
    } = this.props
    const { chooseItemIndex, isSearchArticleModalVisible } = this.state
    return (
      <div className="article-list">
        <div className="list-header">
          {!(selectedMenuKey in EMenuKey) && (
            <div className="list-header-right">
              <RadioGroup
                defaultValue={articlesFilter}
                size="small"
                onChange={this.handleRadioChange}>
                <RadioButton value={EArticleFilter.STARRED}>
                  <Icon type="star" theme="filled" />
                </RadioButton>
                <RadioButton value={EArticleFilter.UNREAD}>
                  <Icon type="file-text" theme="filled" />
                </RadioButton>
                <RadioButton value={EArticleFilter.ALL}>
                  <Icon type="profile" theme="filled" />
                </RadioButton>
              </RadioGroup>
            </div>
          )}
        </div>
        <div className="list-content">
          <ArticleVirtualList scrollToIndex={chooseItemIndex} />
        </div>
        <div className="list-footer">
          <div className="list-footer-left">
            <Icon
              type="check-circle"
              theme="filled"
              className="check-all"
              onClick={this.handleCheckClick}
            />
          </div>
          <div className="list-footer-right">
            <Icon
              type="search"
              className="search-item"
              onClick={this.handleSearchClick}
            />
            {/* <Icon type="right" className="show-content" /> */}
          </div>
        </div>
        <ArticleListSkeleton
          row={8}
          style={{
            display: isFetchingArticles && feeds.size ? 'block' : 'none',
          }}
        />
        <SearchArticleModal
          visible={isSearchArticleModalVisible}
          onCancel={this.handleSearchCancel}
          onItemChoose={this.handleSearchItemChoose}
        />
      </div>
    )
  }
}

export default injectIntl(ArticleListComponent)
