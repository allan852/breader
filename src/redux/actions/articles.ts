import { IArticle, IReduxAction } from '../../schemas'

export const ArticlesActionTypes = {
    SET_ARTICLES: 'SET_ARTICLES',
    SET_ARTICLES_FILTER: 'SET_ARTICLES_FILTER',
    SET_CURRENT_ARTICLE: 'SET_CURRENT_ARTICLE',

    ASYNC_FETCH_ARTICLES: 'ASYNC_FETCH_ARTICLES',
    ASYNC_FILTER_ARTICLES: 'ASYNC_FILTER_ARTICLES',
    ASYNC_SELECT_AND_READ_ARTICLE: 'ASYNC_SELECT_AND_READ_ARTICLE',
    ASYNC_SET_ALL_ARTICLES_READ: 'ASYNC_SET_ALL_ARTICLES_READ',
    ASYNC_STAR_ARTICLE: 'ASYNC_STAR_ARTICLE',
}

export interface ISetCurrentArticlePayload { article: IArticle | null }
export const setCurrentArticleAction = (article: IArticle | null): IReduxAction<ISetCurrentArticlePayload> => ({
    payload: { article },
    type: ArticlesActionTypes.SET_CURRENT_ARTICLE,
})

export interface ISetArticlesFilterPayload { filter: string }
export const setArticlesFilterAction = (filter: string): IReduxAction<ISetArticlesFilterPayload> => ({
    payload: { filter },
    type: ArticlesActionTypes.SET_ARTICLES_FILTER,
})

export interface ISetArticlesPayload { articles: IArticle[]}
export const setArticlesAction = (articles: IArticle[]): IReduxAction<ISetArticlesPayload> => ({
    payload: { articles },
    type: ArticlesActionTypes.SET_ARTICLES,
})

export const asyncFetchArticlesAction = (): IReduxAction<null> => ({
    payload: null,
    type: ArticlesActionTypes.ASYNC_FETCH_ARTICLES,
})

export interface IAsyncFilterArticlesPayload { filter: string }
export const asyncFilterArticlesAction = (filter: string): IReduxAction<IAsyncFilterArticlesPayload> => ({
    payload: { filter },
    type: ArticlesActionTypes.ASYNC_FILTER_ARTICLES,
})

export interface IAsyncSelectAndReadArticlePayload { articleId: string, articleIndex: number }
export const asyncSelectAndReadArticleAction = (articleId: string, articleIndex: number): IReduxAction<IAsyncSelectAndReadArticlePayload> => ({
    payload: { articleId, articleIndex },
    type: ArticlesActionTypes.ASYNC_SELECT_AND_READ_ARTICLE,
})

export interface IAsyncStarArticlePayload { articleId: string, isStarred: boolean }
export const asyncStarArticleAction = (articleId: string, isStarred: boolean): IReduxAction<IAsyncStarArticlePayload> => ({
    payload: { articleId, isStarred },
    type: ArticlesActionTypes.ASYNC_STAR_ARTICLE,
})

export interface IAsyncSetAllArticlesReadPayload { articleIds: string[] }
export const asyncSetAllArticlesReadAction = (articleIds: string[]): IReduxAction<IAsyncSetAllArticlesReadPayload> => ({
    payload: { articleIds },
    type: ArticlesActionTypes.ASYNC_SET_ALL_ARTICLES_READ,
})
