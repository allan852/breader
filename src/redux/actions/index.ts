import { Dispatch } from 'redux'
import { IReduxAction, IReduxAsyncAction } from '../../schemas'

export function asyncActionDispatcher<Type = {}>(dispatch: Dispatch<IReduxAction>, action: IReduxAction) {
    return new Promise<Type>((resolve, reject) => {
        const asyncAction: IReduxAsyncAction = {
            ...action,
            reject,
            resolve,
        }
        dispatch(asyncAction)
    })
}

export * from './articles'
export * from './feeds'
export * from './menu'
