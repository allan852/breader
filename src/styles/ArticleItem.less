@import './Values.less';
.article-item {
    cursor: pointer;
    color: @black-text-color-dark;
    display: flex;
    padding: 4px 8px 4px 8px;
    .item-sider {
        width: 26px;
        flex: 0 0 26px;
    }
    .item-main {
        flex: 1 1 auto;
    }
    .item-header {
        display: block;
        height: 18px;
        line-height: 18px;
        font-size: 12px;
    }
    .item-header-left {
        float: left;
    }
    .item-header-right {
        float: right;
    }
    .item-content {
        font-size: 13px;
        font-weight: 500;
        line-height: 20px;
    }
    &.item-is-unread .item-content {
        color: @black-text-color-light;
    }
    &.item-is-selected {
        background-color: @list-selected-color;
    }
    .item-footer {
        width: 220px;
        overflow: hidden;
        text-overflow: ellipsis;
        white-space: nowrap;
        line-height: 18px;
        font-size: 12px;
    }
}