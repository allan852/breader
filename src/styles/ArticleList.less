@import './Values.less';

.article-list {
  -webkit-user-select: none;
  -webkit-app-region: drag;
  background-color: @beige-background-color;
  position: relative;
  padding: 42px 0 42px 0;
  height: 100%;
  flex: 1 1 auto;

  .list-header {
    .wrapper-header;
    line-height: 42px;
    border-bottom: none;
    padding: 0;
  }
  .list-header-right {
    .wrapper-hf-right;
    .ant-radio-group {
      margin-right: 8px;
    }
    .ant-radio-group-small .ant-radio-button-wrapper {
      padding: 0 11px;
    }
    .ant-radio-button-wrapper {
      color: @black-text-color-dark;
      &:hover {
        color: inherit;
        position: relative;
      }
      &:focus-within {
        outline: none;
      }
    }
    .ant-radio-button-wrapper-checked {
      color: @beige-background-color;
      box-shadow: none;
      border-color: #d9d9d9;
      background-color: @black-text-color-dark;
      &:hover {
        color: @beige-background-color;
      }
    }
  }
  .list-content {
    -webkit-app-region: none;
    height: 100%;
    overflow-x: hidden;
    overflow-y: scroll;
  }
  .list-footer {
    .wrapper-footer;
  }
  .list-footer-left {
    .wrapper-hf-left;
    .anticon {
      margin-top: 13px;
      margin-left: 16px;
    }
  }
  .list-footer-right {
    .wrapper-hf-right;
    .anticon {
      margin-top: 12px;
      margin-right: 16px;
    }
  }
}
