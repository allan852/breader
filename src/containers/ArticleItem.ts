import { connect, MapDispatchToProps, MapStateToProps } from 'react-redux'
import { Dispatch } from 'redux'
import ArticleItem, { IArticleItemOwnProps, IArticleItemReduxDispatch, IArticleItemReduxState } from '../components/ArticleItem'
import { IReduxAction, IReduxState } from '../schemas'

const mapStateToProps: MapStateToProps<IArticleItemReduxState, IArticleItemOwnProps, IReduxState> = (state: IReduxState) => ({
    feedsMap: state.feeds.map,
})

const mapDispatchToProps: MapDispatchToProps<IArticleItemReduxDispatch, IArticleItemOwnProps> = (dispatch: Dispatch<IReduxAction>) => ({
})

export default connect(
    mapStateToProps,
    mapDispatchToProps,
)(ArticleItem)
