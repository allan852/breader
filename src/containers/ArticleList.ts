import { connect, MapDispatchToProps, MapStateToProps } from 'react-redux'
import { Dispatch } from 'redux'
import ArticleList, { IArticleListOwnProps, IArticleListReduxDispatch, IArticleListReduxState } from '../components/ArticleList'
import { asyncActionDispatcher, asyncFilterArticlesAction, asyncSetAllArticlesReadAction } from '../redux/actions'
import { IReduxAction, IReduxState } from '../schemas'

const mapStateToProps: MapStateToProps<IArticleListReduxState, IArticleListOwnProps, IReduxState> = (state: IReduxState) => ({
    articles: state.articles.list,
    articlesFilter: state.articles.filter,
    isFetchingArticles: state.articles.isFetching,
    selectedMenuKey: state.menu.selectedKey,
})

const mapDispatchToProps: MapDispatchToProps<IArticleListReduxDispatch, IArticleListOwnProps> = (dispatch: Dispatch<IReduxAction>) => ({
    asyncFilterArticles: (filter: string) => asyncActionDispatcher(dispatch, asyncFilterArticlesAction(filter)),
    asyncSetAllArticlesRead: (ids: string[]) => asyncActionDispatcher(dispatch, asyncSetAllArticlesReadAction(ids)),
})

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(ArticleList)
