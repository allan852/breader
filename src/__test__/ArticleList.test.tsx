import Enzyme from 'enzyme'
import EnzymeAdapter from 'enzyme-adapter-react-16'
import React from 'react'
import { IntlProvider } from 'react-intl'
import { Provider as ReduxProvider } from 'react-redux'
import TestRenderer from 'react-test-renderer'
import ArticleItem from '../containers/ArticleItem'
import ArticleList from '../containers/ArticleList'
import store from '../redux'
import { setArticlesAction } from '../redux/actions'
import { article, intlProviderProps } from './MockData'

Enzyme.configure({ adapter: new EnzymeAdapter() })

describe('ArticleList Testing', () => {
    const dispatch = store.dispatch

    it('dom testing', () => {
        const wrapper = Enzyme.shallow(
            <ReduxProvider store={store}>
                <IntlProvider {...intlProviderProps}>
                    <ArticleList />
                </IntlProvider>
            </ReduxProvider>
        )
        dispatch(setArticlesAction([article]))
        const instance = wrapper.find(ArticleList)
        // instance.find('.check-all').simulate('click')
        // instance.find('.search-item').simulate('click')
    })

    it('snapshot testing', () => {
        const renderer = TestRenderer.create(
            <ReduxProvider store={store}>
                <IntlProvider {...intlProviderProps}>
                    <ArticleList />
                </IntlProvider>
            </ReduxProvider>
        )
        const tree = renderer.toJSON()
        expect(tree).toMatchSnapshot()
    })
})
