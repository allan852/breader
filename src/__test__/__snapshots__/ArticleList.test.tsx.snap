// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList Testing snapshot testing 1`] = `
<div
  className="article-list"
>
  <div
    className="list-header"
  />
  <div
    className="list-content"
  >
    <div
      className="article-virtual-list"
      onClick={[Function]}
    >
      <div
        style={
          Object {
            "height": 0,
            "overflow": "visible",
            "width": 0,
          }
        }
      >
        <div
          aria-label="grid"
          aria-readonly={true}
          className="ReactVirtualized__Grid ReactVirtualized__List"
          onScroll={[Function]}
          role="grid"
          style={
            Object {
              "WebkitOverflowScrolling": "touch",
              "boxSizing": "border-box",
              "direction": "ltr",
              "height": 0,
              "overflowX": "hidden",
              "overflowY": "auto",
              "position": "relative",
              "width": 0,
              "willChange": "transform",
            }
          }
          tabIndex={0}
        />
      </div>
    </div>
  </div>
  <div
    className="list-footer"
  >
    <div
      className="list-footer-left"
    >
      <i
        aria-label="icon: check-circle"
        className="anticon anticon-check-circle check-all"
        onClick={[Function]}
        tabIndex={-1}
      >
        <svg
          aria-hidden="true"
          className=""
          data-icon="check-circle"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm193.5 301.7l-210.6 292a31.8 31.8 0 0 1-51.7 0L318.5 484.9c-3.8-5.3 0-12.7 6.5-12.7h46.9c10.2 0 19.9 4.9 25.9 13.3l71.2 98.8 157.2-218c6-8.3 15.6-13.3 25.9-13.3H699c6.5 0 10.3 7.4 6.5 12.7z"
          />
        </svg>
      </i>
    </div>
    <div
      className="list-footer-right"
    >
      <i
        aria-label="icon: search"
        className="anticon anticon-search search-item"
        onClick={[Function]}
        tabIndex={-1}
      >
        <svg
          aria-hidden="true"
          className=""
          data-icon="search"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0 0 11.6 0l43.6-43.5a8.2 8.2 0 0 0 0-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
          />
        </svg>
      </i>
    </div>
  </div>
  <div
    className="article-list-skeleton "
    style={
      Object {
        "display": "none",
      }
    }
  >
    <span
      className="span-skeleton skeleton-active "
      style={
        Object {
          "marginLeft": 8,
          "width": 90,
        }
      }
    />
    <div
      className="article-item"
    >
      <div
        className="item-sider"
      >
        <div
          className="avatar-skeleton skeleton-active "
        />
      </div>
      <div
        className="item-main"
      >
        <div
          className="item-header"
        >
          <div
            className="item-header-left"
          >
            <span
              className="span-skeleton skeleton-active "
              style={
                Object {
                  "width": 96,
                }
              }
            />
          </div>
          <div
            className="item-header-right"
          >
            <span
              className="span-skeleton skeleton-active "
              style={
                Object {
                  "width": 36,
                }
              }
            />
          </div>
        </div>
        <div
          className="item-content"
        >
          <div
            className="text-skeleton skeleton-active "
            style={
              Object {
                "width": undefined,
              }
            }
          />
          <div
            className="text-skeleton skeleton-active "
            style={
              Object {
                "width": "67%",
              }
            }
          />
        </div>
        <div
          className="item-footer"
        >
          <span
            className="span-skeleton skeleton-active "
            style={
              Object {
                "width": "100%",
              }
            }
          />
        </div>
      </div>
    </div>
    <div
      className="article-item"
    >
      <div
        className="item-sider"
      >
        <div
          className="avatar-skeleton skeleton-active "
        />
      </div>
      <div
        className="item-main"
      >
        <div
          className="item-header"
        >
          <div
            className="item-header-left"
          >
            <span
              className="span-skeleton skeleton-active "
              style={
                Object {
                  "width": 96,
                }
              }
            />
          </div>
          <div
            className="item-header-right"
          >
            <span
              className="span-skeleton skeleton-active "
              style={
                Object {
                  "width": 36,
                }
              }
            />
          </div>
        </div>
        <div
          className="item-content"
        >
          <div
            className="text-skeleton skeleton-active "
            style={
              Object {
                "width": undefined,
              }
            }
          />
          <div
            className="text-skeleton skeleton-active "
            style={
              Object {
                "width": "67%",
              }
            }
          />
        </div>
        <div
          className="item-footer"
        >
          <span
            className="span-skeleton skeleton-active "
            style={
              Object {
                "width": "100%",
              }
            }
          />
        </div>
      </div>
    </div>
    <div
      className="article-item"
    >
      <div
        className="item-sider"
      >
        <div
          className="avatar-skeleton skeleton-active "
        />
      </div>
      <div
        className="item-main"
      >
        <div
          className="item-header"
        >
          <div
            className="item-header-left"
          >
            <span
              className="span-skeleton skeleton-active "
              style={
                Object {
                  "width": 96,
                }
              }
            />
          </div>
          <div
            className="item-header-right"
          >
            <span
              className="span-skeleton skeleton-active "
              style={
                Object {
                  "width": 36,
                }
              }
            />
          </div>
        </div>
        <div
          className="item-content"
        >
          <div
            className="text-skeleton skeleton-active "
            style={
              Object {
                "width": undefined,
              }
            }
          />
          <div
            className="text-skeleton skeleton-active "
            style={
              Object {
                "width": "67%",
              }
            }
          />
        </div>
        <div
          className="item-footer"
        >
          <span
            className="span-skeleton skeleton-active "
            style={
              Object {
                "width": "100%",
              }
            }
          />
        </div>
      </div>
    </div>
    <div
      className="article-item"
    >
      <div
        className="item-sider"
      >
        <div
          className="avatar-skeleton skeleton-active "
        />
      </div>
      <div
        className="item-main"
      >
        <div
          className="item-header"
        >
          <div
            className="item-header-left"
          >
            <span
              className="span-skeleton skeleton-active "
              style={
                Object {
                  "width": 96,
                }
              }
            />
          </div>
          <div
            className="item-header-right"
          >
            <span
              className="span-skeleton skeleton-active "
              style={
                Object {
                  "width": 36,
                }
              }
            />
          </div>
        </div>
        <div
          className="item-content"
        >
          <div
            className="text-skeleton skeleton-active "
            style={
              Object {
                "width": undefined,
              }
            }
          />
          <div
            className="text-skeleton skeleton-active "
            style={
              Object {
                "width": "67%",
              }
            }
          />
        </div>
        <div
          className="item-footer"
        >
          <span
            className="span-skeleton skeleton-active "
            style={
              Object {
                "width": "100%",
              }
            }
          />
        </div>
      </div>
    </div>
    <div
      className="article-item"
    >
      <div
        className="item-sider"
      >
        <div
          className="avatar-skeleton skeleton-active "
        />
      </div>
      <div
        className="item-main"
      >
        <div
          className="item-header"
        >
          <div
            className="item-header-left"
          >
            <span
              className="span-skeleton skeleton-active "
              style={
                Object {
                  "width": 96,
                }
              }
            />
          </div>
          <div
            className="item-header-right"
          >
            <span
              className="span-skeleton skeleton-active "
              style={
                Object {
                  "width": 36,
                }
              }
            />
          </div>
        </div>
        <div
          className="item-content"
        >
          <div
            className="text-skeleton skeleton-active "
            style={
              Object {
                "width": undefined,
              }
            }
          />
          <div
            className="text-skeleton skeleton-active "
            style={
              Object {
                "width": "67%",
              }
            }
          />
        </div>
        <div
          className="item-footer"
        >
          <span
            className="span-skeleton skeleton-active "
            style={
              Object {
                "width": "100%",
              }
            }
          />
        </div>
      </div>
    </div>
    <div
      className="article-item"
    >
      <div
        className="item-sider"
      >
        <div
          className="avatar-skeleton skeleton-active "
        />
      </div>
      <div
        className="item-main"
      >
        <div
          className="item-header"
        >
          <div
            className="item-header-left"
          >
            <span
              className="span-skeleton skeleton-active "
              style={
                Object {
                  "width": 96,
                }
              }
            />
          </div>
          <div
            className="item-header-right"
          >
            <span
              className="span-skeleton skeleton-active "
              style={
                Object {
                  "width": 36,
                }
              }
            />
          </div>
        </div>
        <div
          className="item-content"
        >
          <div
            className="text-skeleton skeleton-active "
            style={
              Object {
                "width": undefined,
              }
            }
          />
          <div
            className="text-skeleton skeleton-active "
            style={
              Object {
                "width": "67%",
              }
            }
          />
        </div>
        <div
          className="item-footer"
        >
          <span
            className="span-skeleton skeleton-active "
            style={
              Object {
                "width": "100%",
              }
            }
          />
        </div>
      </div>
    </div>
    <div
      className="article-item"
    >
      <div
        className="item-sider"
      >
        <div
          className="avatar-skeleton skeleton-active "
        />
      </div>
      <div
        className="item-main"
      >
        <div
          className="item-header"
        >
          <div
            className="item-header-left"
          >
            <span
              className="span-skeleton skeleton-active "
              style={
                Object {
                  "width": 96,
                }
              }
            />
          </div>
          <div
            className="item-header-right"
          >
            <span
              className="span-skeleton skeleton-active "
              style={
                Object {
                  "width": 36,
                }
              }
            />
          </div>
        </div>
        <div
          className="item-content"
        >
          <div
            className="text-skeleton skeleton-active "
            style={
              Object {
                "width": undefined,
              }
            }
          />
          <div
            className="text-skeleton skeleton-active "
            style={
              Object {
                "width": "67%",
              }
            }
          />
        </div>
        <div
          className="item-footer"
        >
          <span
            className="span-skeleton skeleton-active "
            style={
              Object {
                "width": "100%",
              }
            }
          />
        </div>
      </div>
    </div>
    <div
      className="article-item"
    >
      <div
        className="item-sider"
      >
        <div
          className="avatar-skeleton skeleton-active "
        />
      </div>
      <div
        className="item-main"
      >
        <div
          className="item-header"
        >
          <div
            className="item-header-left"
          >
            <span
              className="span-skeleton skeleton-active "
              style={
                Object {
                  "width": 96,
                }
              }
            />
          </div>
          <div
            className="item-header-right"
          >
            <span
              className="span-skeleton skeleton-active "
              style={
                Object {
                  "width": 36,
                }
              }
            />
          </div>
        </div>
        <div
          className="item-content"
        >
          <div
            className="text-skeleton skeleton-active "
            style={
              Object {
                "width": undefined,
              }
            }
          />
          <div
            className="text-skeleton skeleton-active "
            style={
              Object {
                "width": "67%",
              }
            }
          />
        </div>
        <div
          className="item-footer"
        >
          <span
            className="span-skeleton skeleton-active "
            style={
              Object {
                "width": "100%",
              }
            }
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
